### PR TITLE
Relocate Avatar Spawn Point from the Hips to the Feet

### PR DESCRIPTION
**Pull request by r3tk0n**
Moves the spawn point, via how we set the AddressManager's PositionGetter in Application.cpp, from the Avatar's hips to the 'feet'. This is done via a new function, getWorldFeetPosition(), in Avatar.cpp. This function computes the bottom of the collision capsule.

[Relevant MS ticket.](https://highfidelity.fogbugz.com/f/cases/17249/Relocate-Avatar-Spawn-Point-from-the-Hips-to-the-Feet)

**Test Plan:**
_Goal:_ Verify that teleporting between domains with a large avatar does not cause the avatar to spawn in the floor and pop up.

_Steps:_
- Open Interface
- Equip a large avatar (such as the Barbarian) from the Marketplace.
- Transition between worlds via a teleporter.
- If no 'pop up' occurs (slight falling is acceptable), this PR is good.

**_NOTE:_**
For additional verification that the point that's being calculated is correct, I used the attached test script to spawn an orange circle3d overlay at the calculated position. The circle is centered on this point. To further verify, enable drawing the collision geometry for the avatar in Developer > Avatar > Show Bounding Collision Shapes. You'll notice the very bottom of the lower sphere on the 'capsule' makes contact with drawn overlay, at its tangent.

```"use strict";

(function() {
	var overlayType = "circle3d";
	var overlayColor = { red: 255, green: 128, blue: 0 };
	var overlayPosition = MyAvatar.getWorldFeetPosition();
	var overlayRotation = Quat.angleAxis(90, Vec3.UNIT_X);
	var overlayProps = {
		color: overlayColor,
		visible: true,
		name: "avatarFeetOverlay",
		position: overlayPosition,
		parentID: MyAvatar.sessionUUID,
		filled: true,
		localRotation: overlayRotation
	};
	
	Vec3.print("Adding sphere overlay at: ", overlayPosition);
	var overlayID = Overlays.addOverlay(overlayType, overlayProps);
	
	
	var shutdown = function() {
		Overlays.deleteOverlay(overlayID);
	}
	
	Script.scriptEnding.connect(shutdown);
}());```

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11887/HighFidelity-Beta-PR13764-de5c7f6.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11887/HighFidelity-Beta-PR13764-de5c7f6.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11887/HighFidelity-Beta-Interface-PR13764-de5c7f6.dmg

File: libraries/avatars-renderer/src/avatars-renderer/Avatar.h [hunk 1]
```diff
@@ -249,6 +249,12 @@ class Avatar : public AvatarData, public scriptable::ModelProvider {
     virtual void computeShapeInfo(ShapeInfo& shapeInfo);
     void getCapsule(glm::vec3& start, glm::vec3& end, float& radius);
     float computeMass();
+    /**jsdoc
+     * Get the position of the current avatar's feet (or rather, bottom of its collision capsule) in world coordinates.
+     * @function MyAvatar.getWorldFeetPosition()
```
Comment (ctrlaltdavid) [inline libraries/avatars-renderer/src/avatars-renderer/Avatar.h]: Shouldn't have "()" after the function name.

File: libraries/avatars-renderer/src/avatars-renderer/Avatar.h [hunk 2]
```diff
@@ -249,6 +249,12 @@ class Avatar : public AvatarData, public scriptable::ModelProvider {
...
+     * @returns {Vec3}
```
Comment (ctrlaltdavid) [inline libraries/avatars-renderer/src/avatars-renderer/Avatar.h]: Should have a description of the return value.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11894/HighFidelity-Beta-PR13764-d04ac2b.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11894/HighFidelity-Beta-PR13764-d04ac2b.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11894/HighFidelity-Beta-Interface-PR13764-d04ac2b.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11895/HighFidelity-Beta-PR13764-b590ff7.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11895/HighFidelity-Beta-PR13764-b590ff7.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11895/HighFidelity-Beta-Interface-PR13764-b590ff7.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11895/HighFidelity-Beta-PR13764-b590ff7.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11895/HighFidelity-Beta-Interface-PR13764-b590ff7.exe

File: libraries/avatars-renderer/src/avatars-renderer/Avatar.h [hunk 3]
```diff
@@ -251,8 +251,8 @@ class Avatar : public AvatarData, public scriptable::ModelProvider {
     float computeMass();
     /**jsdoc
      * Get the position of the current avatar's feet (or rather, bottom of its collision capsule) in world coordinates.
-     * @function MyAvatar.getWorldFeetPosition()
-     * @returns {Vec3}
+     * @function MyAvatar.getWorldFeetPosition
+     * @returns {Vec3} The position of the Avatar's feet in world coordinates.
```
Comment (wayne-chen) [inline libraries/avatars-renderer/src/avatars-renderer/Avatar.h]: I think `Avatar` should be lowercase. :)

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11897/HighFidelity-Beta-PR13764-1aa9fcd.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11897/HighFidelity-Beta-PR13764-1aa9fcd.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11897/HighFidelity-Beta-Interface-PR13764-1aa9fcd.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11897/HighFidelity-Beta-PR13764-1aa9fcd.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11897/HighFidelity-Beta-Interface-PR13764-1aa9fcd.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11924/HighFidelity-Beta-PR13764-32165df.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11924/HighFidelity-Beta-PR13764-32165df.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11924/HighFidelity-Beta-Interface-PR13764-32165df.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11924/HighFidelity-Beta-PR13764-32165df.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11924/HighFidelity-Beta-Interface-PR13764-32165df.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Pull request merged**